pkg/db/src/implementations/version-1.2/stock.ts

```diff
@@ -51,11 +52,20 @@ class Stock implements StockInterface {
 	async query() {
 		const queryRes = await this.#db._pouch.allDocs({ ...this.options, include_docs: true });
 
+		const warehouses = wrapIter(queryRes.rows)
+			.map(({ doc }) => doc as Doc)
+			.filter((doc): doc is WarehouseData => doc?.docType === "warehouse")
+			.array();
+
 		const mapGenerator = wrapIter(queryRes.rows)
 			.map(({ doc }) => doc as Doc)
 			.filter((doc): doc is NoteData => doc?.docType === "note")
 			.filter(({ committed, entries }) => Boolean(committed && entries?.length))
-			.flatMap(({ entries, noteType }) => wrapIter(entries).map((entry) => ({ ...entry, noteType })));
+			.flatMap(({ entries, noteType }) =>
+				wrapIter(entries)
+					.filter((entry) => warehouses.some((warehouse) => warehouse._id === entry.warehouseId))
+					.map((entry) => ({ ...entry, noteType }))
+			);
```

Comment:
I have a hunch this will bugs

Reply:
Ok, scratch that ☝️ it will work, but let's solve the problem at the root:

Can we move the fix even further upstream: have the `queryRes` no include entries belonging to non-existing warehouses?